src/System.Net.Http/src/System/Net/Http/Unix/CurlResponseMessage.cs

```diff
@@ -0,0 +1,247 @@
+            {
+                get
+                {
+                    CheckDisposed();
+                    throw new NotSupportedException();
+                }
+
+                set
+                {
+                    CheckDisposed();
+                    throw new NotSupportedException();
+                }
+            }
+
+            public override void Flush()
+            {
+                // Nothing to do.
+            }
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                if (buffer == null)
+                {
+                    throw new ArgumentNullException("buffer");
+                }
+
+                if (offset < 0)
+                {
+                    throw new ArgumentOutOfRangeException("offset");
+                }
+
+                if (count < 0)
+                {
+                    throw new ArgumentOutOfRangeException("count");
+                }
+
+                if ((long) offset + (long) count > (long) buffer.Length)
+                {
+                    throw new ArgumentException("buffer");
+                }
+
+                CheckDisposed();
+
+                do
+                {
+                    lock (_lockObject)
+                    {
+                        // Wait till data is available
+                        if (_innerStream == null)
+                        {
+                            SignalWriter();
+                            continue;
+                        }
+
+                        int bytesRead = _innerStream.Read(buffer, offset, (int)Math.Min((long)count, _innerStream.Length));
+                        if (_innerStream.Position == _innerStream.Length)
+                        {
+                            _innerStream.Dispose();
+                            _innerStream = null;
+                            _noMoreDataEvent.Set();
+                        }
+                        return bytesRead;
+                    }
+                }
+                while (WaitForWriter());
+                return 0;
+            }
+
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                CheckDisposed();
+                throw new NotSupportedException();
+            }
+
+            public override void SetLength(long value)
+            {
+                CheckDisposed();
+                throw new NotSupportedException();
+            }
+
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                CheckDisposed();
+                throw new NotSupportedException();
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                if (disposing && !_disposed)
+                {
+                    // Cancel any pending operations
+                    SignalComplete(true);
+
+                    _disposed = true;
+                    if (_innerStream != null)
```

Comment:
This state is only ever accessed while holding the lock on lockObject.  It's ok not to do so here?